masterdata_context.md

```diff
@@ -2,34 +2,36 @@
+Type | GREM Status | REM Status | Description
+---|---|---|---
+[Masterdata.Property.Created](#masterdatapropertycreated) | | Draft 9.2 | A new property has been created
+[Masterdata.Property.Updated](#masterdatapropertyupdated) | | Draft 9.2 | Data associated to a property has changed; you get changed attributes only
+[Masterdata.Property.TagAdded](#masterdatapropertytagadded) | | N/A | A tag was added to a property; please read the specs for this event carefully
+[Masterdata.Property.TagRemoved](#masterdatapropertytagremoved) | | N/A | A tag was removed from a property; please read the specs for this event carefully
+[Masterdata.Building.Created](#masterdatabuildingcreated) | | Draft 9.2 | A building has been created
+[Masterdata.Building.Updated](#masterdatabuildingupdated) | | Draft 9.2 | Data associated to a building has changed; you get the reference plus all changed attributes
+[Masterdata.Building.Deleted](#masterdatabuildingdeleted) | | Draft 9.2 | The building was deleted
+[Masterdata.Unit.Created](#masterdataunitcreated) | | Draft 9.2 | A rentable unit has been created
+[Masterdata.Unit.Updated](#masterdataunitupdated) | | Draft 9.2 | Data associated to a rentable unit has changed; you get the reference plus all changed attributes
+[Masterdata.Unit.Deleted](#masterdataunitdeleted) | | Draft 9.2 | The unit was deleted
+[Masterdata.Unit.ReferenceChanged](#masterdataunitreferencechanged) | | Draft 9.2 | The unit reference has changed
+[Masterdata.ManagementTeam.Updated](#masterdatamanagementteamupdated) | | N/A | A change to a property management team was applied; only changed roles are published
+[Masterdata.Configuration.SedexIdChanged](#masterdataconfigurationsedexidchanged) | Draft | 9.1 | A new SedexID has been configured |
 
 ### Masterdata.Property.Created
 
-Field | Type | Content / Remarks
----|---|---
-eventType | string | Masterdata.Property.Created
-data | hash |
-&nbsp;&nbsp;reference | string | unique identifier for the property
-&nbsp;&nbsp;description | string |
-&nbsp;&nbsp;zipCode | string |
-&nbsp;&nbsp;city | string |
-&nbsp;&nbsp;countryCode | string | ISO country code, eg 'CH'
-&nbsp;&nbsp;mandateTerminatedBy | string | ISO date, eg '2018-12-31'
+Field | Type | GREM Status | REM Status | Content / Remarks
+---|---|---|---|---
+eventType | string | | Draft 9.2 | Masterdata.Property.Created
+data | hash | | |
+&nbsp;&nbsp;reference | string | | Draft 9.2 | unique identifier for the property
+&nbsp;&nbsp;description | string | | Draft 9.2 |
+&nbsp;&nbsp;zipCode | string | | Draft 9.2 |
+&nbsp;&nbsp;city | string | | Draft 9.2 |
+&nbsp;&nbsp;countryCode | string | | Draft 9.2 | ISO country code, eg 'CH'
+&nbsp;&nbsp;mandateTerminatedBy | string | | N/A | ISO 8601 encoded date, eg '2019-03-01'
+&nbsp;&nbsp;startOfAdministration | string | Draft | Draft 9.2 | ISO 8601 encoded date, eg '2019-03-01'
```

Comment:
Aus Sicht MPE-Serivce braucht ein Created-Event keine (neuen) Properties da der Service mit einem GraphQL-Query auf Created reagieren wird. So kann der Initalload und dieser Fall konzeptionell gleich implementiert werden. Diese gilt auch fuer alle unten aufgefuehrten Created-Events.

Reply:
It is actually a rather formal question: what is the meaning of "Initial Load"?
* If the initial load is the process of retrieving data from the GraphQL-Backend, then yes.
* If the initial load is what a service does at the very beginning of its lifetime, and if it gets every further delta (including newly created objects) through messaging, then no.

Because of all discrepancies between the GraphQL-schema and the messages, it is a bit hard to say.

Reply:
The term "initial load" was introduced when I was designing the first iteration of our API Layer together with Allthings. It means "initial setup of the local data model with current data".

Just as a reminder: All those Created-Events are already in production and used by several external services since months. GARAIO REM will send all these attributes because that's how we sell our API layer. An external service should never have to make GraphQL queries to get additional data about an item, his local data model should be updated just by applying the message attributes.